### PR TITLE
Use tagged version of CRDB, pin toolchain, ignore broken tests

### PR DESCRIPTION
**Pull request by smklein**
Performing whatever changes are necessary to make CI green

### CODE REVIEW

**Comment by smklein** [comment]
The error:

```
error: could not compile `oxide`
Caused by:
  process didn't exit successfully: `rustc --crate-name oxide --edition=2021 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test -C metadata=859b049955092ec9 -C extra-filename=-859b049955092ec9 --out-dir /home/runner/work/cli/cli/target/debug/deps -C incremental=/home/runner/work/cli/cli/target/debug/incremental -L dependency=/home/runner/work/cli/cli/target/debug/deps --extern inflector=/home/runner/work/cli/cli/target/debug/deps/libinflector-13671c413d8ea0f0.rlib --extern ansi_term=/home/runner/work/cli/cli/target/debug/deps/libansi_term-45f409e7b3447356.rlib --extern anyhow=/home/runner/work/cli/cli/target/debug/deps/libanyhow-564ef1f325a14056.rlib --extern async_trait=/home/runner/work/cli/cli/target/debug/deps/libasync_trait-85a92e262de869ce.so --extern atty=/home/runner/work/cli/cli/target/debug/deps/libatty-d68fa8358df55a37.rlib --extern byte_unit=/home/runner/work/cli/cli/target/debug/deps/libbyte_unit-3858cda81df0adb1.rlib --extern chrono=/home/runner/work/cli/cli/target/debug/deps/libchrono-6b85f15861dbdc33.rlib --extern chrono_humanize=/home/runner/work/cli/cli/target/debug/deps/libchrono_humanize-73b8f499e3a3a849.rlib --extern clap=/home/runner/work/cli/cli/target/debug/deps/libclap-83a239d8d4b3515c.rlib --extern clap_complete=/home/runner/work/cli/cli/target/debug/deps/libclap_complete-c3c760df2f495aac.rlib --extern cli_macro=/home/runner/work/cli/cli/target/debug/deps/libcli_macro-d0fae4c13e7d9132.so --extern colored_json=/home/runner/work/cli/cli/target/debug/deps/libcolored_json-47824666916df24d.rlib --extern data_encoding=/home/runner/work/cli/cli/target/debug/deps/libdata_encoding-d418946c8be0ca98.rlib --extern dialoguer=/home/runner/work/cli/cli/target/debug/deps/libdialoguer-dbe9134a8d94baee.rlib --extern dirs=/home/runner/work/cli/cli/target/debug/deps/libdirs-5e337363ddb45f58.rlib --extern futures=/home/runner/work/cli/cli/target/debug/deps/libfutures-57b88cd343856e14.rlib --extern git_rev=/home/runner/work/cli/cli/target/debug/deps/libgit_rev-4d98191fb16dd2a4.so --extern heck=/home/runner/work/cli/cli/target/debug/deps/libheck-101bfbbdccd014e6.rlib --extern http=/home/runner/work/cli/cli/target/debug/deps/libhttp-072ce2e4f15486cd.rlib --extern ipnetwork=/home/runner/work/cli/cli/target/debug/deps/libipnetwork-c3888298cca369e2.rlib --extern log=/home/runner/work/cli/cli/target/debug/deps/liblog-27e48d87ddc72cce.rlib --extern num_traits=/home/runner/work/cli/cli/target/debug/deps/libnum_traits-26cb3395736bf7b3.rlib --extern open=/home/runner/work/cli/cli/target/debug/deps/libopen-97e8172de76a7f88.rlib --extern oxide_api=/home/runner/work/cli/cli/target/debug/deps/liboxide_api-ab8aae470f43faa2.rlib --extern parse_display=/home/runner/work/cli/cli/target/debug/deps/libparse_display-1b1d7fab0a531450.rlib --extern pretty_assertions=/home/runner/work/cli/cli/target/debug/deps/libpretty_assertions-05b7c9b016369cba.rlib --extern pulldown_cmark=/home/runner/work/cli/cli/target/debug/deps/libpulldown_cmark-4b10093bd0e98d71.rlib --extern pulldown_cmark_to_cmark=/home/runner/work/cli/cli/target/debug/deps/libpulldown_cmark_to_cmark-c1b48d28c576e855.rlib --extern regex=/home/runner/work/cli/cli/target/debug/deps/libregex-adc62d18636202cb.rlib --extern reqwest=/home/runner/work/cli/cli/target/debug/deps/libreqwest-5707a4651a6877c8.rlib --extern ring=/home/runner/work/cli/cli/target/debug/deps/libring-0b3f5fb49ca0c988.rlib --extern roff=/home/runner/work/cli/cli/target/debug/deps/libroff-ef811cb22157c55c.rlib --extern serde=/home/runner/work/cli/cli/target/debug/deps/libserde-8ac462207d654c70.rlib --extern serde_json=/home/runner/work/cli/cli/target/debug/deps/libserde_json-523958630db28af4.rlib --extern serde_yaml=/home/runner/work/cli/cli/target/debug/deps/libserde_yaml-35386e059fa0769c.rlib --extern serial_test=/home/runner/work/cli/cli/target/debug/deps/libserial_test-ac4b584dc9bf18f4.rlib --extern sha2=/home/runner/work/cli/cli/target/debug/deps/libsha2-f6db679707793e28.rlib --extern shlex=/home/runner/work/cli/cli/target/debug/deps/libshlex-680ff170ce04ba42.rlib --extern slog=/home/runner/work/cli/cli/target/debug/deps/libslog-8c26b0b060166927.rlib --extern slog_async=/home/runner/work/cli/cli/target/debug/deps/libslog_async-9a772546624cba8b.rlib --extern slog_scope=/home/runner/work/cli/cli/target/debug/deps/libslog_scope-228038310e9cb236.rlib --extern slog_stdlog=/home/runner/work/cli/cli/target/debug/deps/libslog_stdlog-6e5b2231c526bcac.rlib --extern slog_term=/home/runner/work/cli/cli/target/debug/deps/libslog_term-24c3d7f8a33ecb61.rlib --extern sshkeys=/home/runner/work/cli/cli/target/debug/deps/libsshkeys-f2c03f67d868743d.rlib --extern tabled=/home/runner/work/cli/cli/target/debug/deps/libtabled-281bd780274f85cc.rlib --extern tabwriter=/home/runner/work/cli/cli/target/debug/deps/libtabwriter-5f84c6cef19782c1.rlib --extern tempfile=/home/runner/work/cli/cli/target/debug/deps/libtempfile-2cdec437d0df1b01.rlib --extern termbg=/home/runner/work/cli/cli/target/debug/deps/libtermbg-2542c34a9379ab22.rlib --extern terminal_spinners=/home/runner/work/cli/cli/target/debug/deps/libterminal_spinners-0e5ee8194d80c989.rlib --extern terminal_size=/home/runner/work/cli/cli/target/debug/deps/libterminal_size-f7dab9109211ee5e.rlib --extern test_context=/home/runner/work/cli/cli/target/debug/deps/libtest_context-991aa441478267f3.rlib --extern thiserror=/home/runner/work/cli/cli/target/debug/deps/libthiserror-a01b3b0a34197443.rlib --extern tokio=/home/runner/work/cli/cli/target/debug/deps/libtokio-4b1b2a8e2205a8e2.rlib --extern toml=/home/runner/work/cli/cli/target/debug/deps/libtoml-f9e262f9fca43644.rlib --extern toml_edit=/home/runner/work/cli/cli/target/debug/deps/libtoml_edit-36a21e7dc2bf4ad5.rlib --extern url=/home/runner/work/cli/cli/target/debug/deps/liburl-ca43c6ddadd3edd0.rlib --extern uuid=/home/runner/work/cli/cli/target/debug/deps/libuuid-181538278acc2662.rlib --extern version_compare=/home/runner/work/cli/cli/target/debug/deps/libversion_compare-983f6c22994c304a.rlib -L native=/home/runner/work/cli/cli/target/debug/build/ring-92e5bb2bbc96a21b/out` (signal: 9, SIGKILL: kill)
```

Looks like the system is running out of memory and OOM-ing

**Comment by smklein** [comment]
Notably, when doing `cargo build`, I see my system using ~1-3 GiB of memory for rustc. Doing `cargo test` -- while still compiling -- I see this bump to over 20 GiB of memory consumed by rustc.

**Comment by smklein** [comment]
This issue can be trivially reproduced with `cargo build --tests`

**Comment by smklein** [comment]
FWIW, I am seeing this RAM consumption with `rustc 1.61.0 (fe5b13d68 2022-05-18)` - and also on CI. Using `rustc 1.64.0-nightly (c80c4b8fd 2022-06-26)`, I no longer see this issue.

Attempting to pin the toolchain to nightly as a workaround.

**Comment by david-crespo** [comment]
Seems like it got past the GRANT error!

**Comment by smklein** [comment]
At this point, the tests are failing, but we're back in "tests are failing normally" land, rather than setup failing / OOMs.

**Comment by david-crespo** [comment]
Miles ahead of where we were. And they only took about 6 minutes to fail — that's pretty good.

**Comment by karencfv** [comment]
Woohoo!!!! This is so great!!! Thank a bunch for the help 🙇‍♀️ At this point the errors seem to be related with the CLI not logging in for some reason, but that's unrelated to the simulated omicron set up. Thanks so much for your help @smklein, I know you're super busy, so I won't take more of your time. I can take it up from here!

If it's OK with you I'd like to merge this PR, unless you think there's something missing?

**Comment by david-crespo** [comment]
Random thought, how about commenting out or disabling the failing tests so we have a passing CI build to work from?

**Comment by smklein** [comment]
I'm using the `#[ignore]` macro, which still compiles them, but otherwise does the same.

Filed (and referenced) both https://github.com/oxidecomputer/cli/issues/204 and https://github.com/oxidecomputer/cli/issues/205

**Comment by karencfv** [comment]
Just ran the tests locally with nightly toolchain. Looks like that's the reason they fail 😅 

So, we're in a situation where we need nightly for the system to not run out of memory, but we need stable to get the tests actually running 🙃

**Comment by smklein** [comment]
> Just ran the tests locally with nightly toolchain. Looks like that's the reason they fail sweat_smile
> 
> So, we're in a situation where we need nightly for the system to not run out of memory, but we need stable to get the tests actually running upside_down_face

Is this true if you check out this PR? And how are you running tests?

Locally, nightly appears to work fine for me - I pinned the toolchain in the `rust-toolchain.toml` file in this file to try to force the same for everyone.

**Comment by karencfv** [comment]
Huh, that's odd. I'm just doing a `cargo test` against a simulated omicron. When I use stable it works, when I use nightly it gives me those errors

**Comment by karencfv** [comment]
will try again with your changes to the toolchain

**Comment by karencfv** [comment]
OMG THE TESTS PASSED!!!!!!!!!!

File: src/tests.rs [hunk 1]
```diff
@@ -103,7 +103,9 @@ impl AsyncTestContext for MainContext {
     }
 }
 
+// TODO(https://github.com/oxidecomputer/cli/issues/204): Fix this test.
 #[test_context(MainContext)]
+#[ignore]
```
Comment (smklein) [inline src/tests.rs:108]: FWIW, this is causing the major loss in coverage, this is actually a wrapper around many more tests.

Executing the test, I see:

```rust
thread 'tests::test_main' panicked at 'failed to get racks: Internal Error: unable to parse query string: unknown variant `id_ascending`, expected `id-ascending`', src/tests.rs:38:14
```

Seems like it should be relatively to fix by updating the `oxide-api`.

Comment (karencfv) [reply src/tests.rs:108]: Totally fine! You've already done so much!!!! Thanks again 🙇‍♀️ 🎉

**Review by karencfv — APPROVED** [review]
🙇‍♀️

**Pull request merged**